src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java

```diff
@@ -139,6 +139,10 @@ public QueryByUIFilterGroup(IForm form, FilterGroup filterGroup, CompletionFlag
 
         switch (fem.getElementType()) {
           case BOOLEAN:
+          case INTEGER:
+          case DECIMAL:
+          case SELECT1:
+          case STRING:
```

Comment:
Much cleaner!

Reply:
Yup, you have an eye for this :) Thanks!